libs/langchain/langchain/storage/redis.py

```diff
@@ -1,7 +1,10 @@
-from typing import Any, Iterator, List, Optional, Sequence, Tuple, cast
+from typing import TYPE_CHECKING, Iterator, List, Optional, Sequence, Tuple, cast
 
 from langchain.schema import BaseStore
 
+if TYPE_CHECKING:
```

Comment:
@baskaryan what's our approach for type hinting optional code that is on the unit test path?

Reply:
does this cause issues? wouldn't it skip this during unit test

Reply:
think you just need a 
```python
from __future__ import annotations
```
 in there

Reply:
Added other stuff to this PR and removed the type check for now -- can follow up in a bit